spec/user_interface_spec.rb

```diff
@@ -72,4 +72,61 @@
 
     expect(ui.run).to eq(1)
   end
+
+  describe 'ask for fields method' do
+    let(:input) { StringIO.new(contact_inputs(TEST_DETAILS)) }
+    let(:ui) { described_class.new(input, output) }
+
+    it 'asks user for all fields' do
+      ui.ask_for_fields
+
+      expect(output.string).to include(described_class::FIELDS_TO_PROMPTS.values.join("\n"))
+    end
+
+    it 'reads the input for all prompt' do
+      user_input = ui.ask_for_fields
+
+      expect(user_input).to eq(TEST_DETAILS)
+    end
+
+    it 'sends error if number is invalid' do
```

Comment:
`'prints error'` rather than `'sends error'`

this is misleading as we are not sending the error anywhere, we are just printing it

also `number` is very generic, you mean `phone` here.